src/frontend/Typechecker.ml

```diff
@@ -1079,54 +1079,49 @@ let verify_valid_sampling_pos loc cf =
 let verify_sampling_distribution loc tenv id arguments =
   let name = id.name in
   let argumenttypes = List.map ~f:arg_type arguments in
-  let is_name_w_suffix_sampling_dist suffix =
-    match
-      SignatureMismatch.matching_stanlib_function (name ^ suffix) argumenttypes
-    with
-    | UniqueMatch (ReturnType UReal, _, _) -> true
-    | _ -> false in
-  let is_sampling_dist_in_math =
-    List.exists ~f:is_name_w_suffix_sampling_dist
-      (Utils.distribution_suffices @ Utils.unnormalized_suffices)
+  let name_w_suffix_sampling_dist suffix =
+    SignatureMismatch.matching_function tenv (name ^ suffix) argumenttypes in
+  let sampling_dists =
+    List.map ~f:name_w_suffix_sampling_dist
+      (Utils.distribution_suffices @ Utils.unnormalized_suffices) in
```

Comment:
As far as I know `matching_function` standardizes the suffix before name lookup so `distribution_suffices` already covers all functions here and `unnormalized_suffices` just adds duplicates to the list.

Reply:
Huh, you're right. I was just following existing behavior here, which means it's been doing extra work for who-knows how long